File: src/main/java/org/hisp/dhis/rules/models/RuleValueType.java
package org.hisp.dhis.rules.models;

import javax.annotation.Nonnull;

public enum RuleValueType
{
    TEXT( "" ), NUMERIC( 0d ), BOOLEAN( false ), DATE( "2020-01-01" );

    @Nonnull
    private final Object defaultValue;

    RuleValueType( @Nonnull Object defaultValue )
    {
        this.defaultValue = defaultValue;
    }

    @Nonnull
    public Object defaultValue()
    {
        return defaultValue;
    }
}

File: src/main/java/org/hisp/dhis/rules/models/RuleValueType.kt
package org.hisp.dhis.rules.models

enum class RuleValueType(private val defaultValue: Any) {
    TEXT(""),
    NUMERIC(0.0),
    BOOLEAN(false),
    DATE("2020-01-01");

    fun defaultValue(): Any {
        return defaultValue
    }
}
